elements/lisk-p2p/README.md

```diff
@@ -1,13 +1,102 @@
+
+const p2p = new P2P({
+	nodeInfo: {
+		wsPort: 5001,
+		nethash: '123xyz', // network identifier
+		protocolVersion: '1.1',
+	},
+});
+
+const run = async () => {
+	await p2p.start();
+};
+
+run()
+	.then(() => console.log('P2P node has started!'))
+	.catch(err => {
+		console.log(`Error occurred while running p2p node: ${err}`);
+		p2p.stop();
+		process.exit(1);
+	});
+```
+
+It will start a node that is available on `5001` websocket port.
+
+```shell
+P2P node has started!
+```
+
+### Actions
+
+It provides simple interface to send, request, broadcast information and many more functions to interact with the network.
+
+- `p2p.start()` to start a P2P node after creating an instance.
+- `p2p.stop()` to stop a P2P node.
+- `p2p.config` to get the config of the node.
+- `p2p.isActive` to check the status if the node is up and running.
+- `p2p.nodeInfo` to check the node status and information.
+- `applyNodeInfo(nodeInfo: P2PNodeInfo)` to broadcast your updated `nodeInfo` to the network.
+- `p2p.getConnectedPeers()` to get all the connected peers that are connected to your node in the network.
+- `p2p.getDisconnectedPeers` to get all the disconnected peers that are part of the network but not connected to you.
+- `p2p.request(packet: P2PRequestPacket)` to request information from the network that will run the peer selection and finds an appropriate peer for you to request information.
+- `p2p.send(message: P2PMessagePacket)` it will send the information to 16 connected peers choosen by peer selection for send.
+- `broadcast(message: P2PMessagePacket)` to broadcast information to all the connected peers.
+- `requestFromPeer(packet: P2PRequestPacket,peerId: string)` to request from a specific peers in the network.
+- `sendToPeer(message: P2PMessagePacket, peerId: string)` to send information to a specific peer in the connected peers.
+
+### Events
+
+We can listen to various events on the network to observe the network activities more closely and take appropriate actions if needed.
+
+```typescript
+p2p.on(EVENT_CONNECT_OUTBOUND, peerInfo => {
+	// Take any action based on outbound connect event
+});
+```
+
+#### Available events
+
+- `EVENT_BAN_PEER` - When a peer is banned
+- `EVENT_CLOSE_INBOUND`
+- `EVENT_CLOSE_OUTBOUND`
+- `EVENT_CONNECT_ABORT_OUTBOUND`
+- `EVENT_CONNECT_OUTBOUND`
+- `EVENT_DISCOVERED_PEER`
+- `EVENT_FAILED_PEER_INFO_UPDATE`
+- `EVENT_FAILED_TO_ADD_INBOUND_PEER`
+- `EVENT_FAILED_TO_COLLECT_PEER_DETAILS_ON_CONNECT`
+- `EVENT_FAILED_TO_FETCH_PEER_INFO`
+- `EVENT_FAILED_TO_FETCH_PEERS`
+- `EVENT_FAILED_TO_PUSH_NODE_INFO`
+- `EVENT_FAILED_TO_SEND_MESSAGE`
+- `EVENT_INBOUND_SOCKET_ERROR`
+- `EVENT_MESSAGE_RECEIVED`
+- `EVENT_NETWORK_READY`
+- `EVENT_NEW_INBOUND_PEER`
+- `EVENT_OUTBOUND_SOCKET_ERROR`
+- `EVENT_REMOVE_PEER`
+- `EVENT_REQUEST_RECEIVED`
+- `EVENT_UNBAN_PEER`
+- `EVENT_UPDATED_PEER_INFO`
+
+### Examples
+
+Check it under `lisk-p2p/examples` folder for a few examples to demonstrate P2P library usage and some use cases.
```

Comment:
Suggestion: List the different examples and explain what each example will learn the "student". Then we can link each example to the right code file/folder.

Reply:
good suggestion: just updated it 👍